plugins/woocommerce/tests/e2e-pw/tests/activate-and-setup/complete-onboarding-wizard.spec.js

```diff
@@ -227,8 +227,8 @@ test.describe(
 				page.locator( 'a:has-text("WooCommerce Payments")' )
 			).toHaveCount( 0 );
 			// Uncheck all business features
-			if ( page.isChecked( '#inspector-checkbox-control-1' ) ) {
-				await page.click( '#inspector-checkbox-control-1' );
+			if ( page.isChecked( '.components-checkbox-control__input' ) ) {
+				await page.click( '.components-checkbox-control__input' );
```

Comment:
I think this will match the first checkbox on the page. This should work ok for now, but might be flaky in the future if another checkbox is added elsewhere on the page before it.

There doesn't seem like a way we can match the checkbox precisely at the moment. It's missing an associated `label` so we can't do a text match either. That's probably an accessibility concern as well.

I'll create a follow-up issue to make this checkbox easier to match.

Reply:
Thanks @adrianduffell!

The good practices suggest using the custom [`data-*` attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/data-*) to access elements for test automation, such as `data-test` or `data-test-id`, for example, as requested in the following [Selenium PR](https://github.com/SeleniumHQ/selenium-ide/issues/435) or in the [good practices recommended by Cypress](https://docs.cypress.io/guides/references/best-practices#Selecting-Elements).

The reason for this is that even if we use a specific `className` or `id` of the element in our tests, these might change as part of a refactor and the test would need to be updated. On the other hand, if we use custom `data-*` locators in our tests, these locators will be specifically created for this purpose, so we would decouple the automated tests from the front-end implementation details.

It would be great if we could start adding these `data-*` custom attributes to elements from now on 🔥

Reply:
@alopezari I'd agree, that would be great 🔥 

Unfortunately the underlying WordPress/Gutenberg [CheckboxControl component](https://github.com/WordPress/gutenberg/tree/a3c753d28ddb0972311e2da0136277626bdf832e/packages/components/src/checkbox-control) doesn't allow us to configure it with data attributes, but maybe we could try to move forward with a PR in their repo to bring in support for it 🤔

Reply:
Oh, I didn't know about that, thanks for letting me know @adrianduffell!

> but maybe we could try to move forward with a PR in their repo to bring in support for it
That could be quite interesting. And if that PR becomes successful, that would leave the door open to do the same with other components 😲

Reply:
> I think this will match the first checkbox on the page. This should work ok for now, but might be flaky in the future if another checkbox is added elsewhere on the page before it.
> 
> There doesn't seem like a way we can match the checkbox precisely at the moment. It's missing an associated label so we can't do a text match either. That's probably an accessibility concern as well.

Yes agreed, I think switching to the class for selector isn't ideal but is the only way to improve the selector for now.

> I'll create a follow-up issue to make this checkbox easier to match.

Thank you!